src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -137,22 +138,42 @@ private sealed class LaunchCompleteCallback(
         IProjectThreadingService threadingService,
         DebugLaunchOptions launchOptions,
         IDebugProfileLaunchTargetsProvider? targetsProvider,
-        ILaunchProfile activeProfile)
-        : IVsDebuggerLaunchCompletionCallback
+        ILaunchProfile activeProfile,
+        IReadOnlyList<IDebugLaunchSettings> debugLaunchSettings)
+        : IVsDebuggerLaunchCompletionCallback, IVsDebugProcessNotify1800
     {
+        private readonly List<IVsLaunchedProcess> _launchedProcesses = new();
         public void OnComplete(int hr, uint debugTargetCount, VsDebugTargetProcessInfo[] processInfoArray)
         {
             ErrorHandler.ThrowOnFailure(hr);
 
             if (targetsProvider is IDebugProfileLaunchTargetsProvider4 targetsProvider4)
             {
                 threadingService.ExecuteSynchronously(() => targetsProvider4.OnAfterLaunchAsync(launchOptions, activeProfile, processInfoArray));
+
+                if (targetsProvider4 is IDebugProfileLaunchTargetsProvider5 targetsProvider5 &&
+                    debugLaunchSettings.Count == 1 &&
+                    processInfoArray.Length == 1 &&
+                    _launchedProcesses.Count == 1)
+                {
+                    threadingService.ExecuteSynchronously(() => targetsProvider5.OnAfterLaunchAsync(launchOptions, activeProfile, debugLaunchSettings[0], _launchedProcesses[0], processInfoArray[0]));
+                }
```

Comment:
Looking at this again I noticed that `_launchedProcesses` is protected by a `lock` in other places, but not here. Is it possible that `_launchedProcesses` is cleared between the check for the single item and the dereferencing of that item? If so, this could hit an array index exception.